doc/source/rllib/connector-v2.rst

```diff
@@ -0,0 +1,596 @@
+                high=list(in_obs_space.high) + [float("inf")],
+                shape=(in_obs_space.shape[0] + 1,),
+                dtype=in_obs_space.dtype,
+            )
+
+        # Define the actual transformation.
+        def __call__(self, *, rl_module, batch, episodes, **kwargs):
+            # Loop through all `episodes`.
+            for single_agent_episode in self.single_agent_episode_iterator(episodes):
+                # Get last reward (or 0.0 if right after `env.reset`).
+                reward = single_agent_episode.get_reward(-1, fill=0.0)
+                # Get last observation.
+                obs = single_agent_episode.get_observation(-1)
+                # Append reward to obs array.
+                new_obs = np.append(obs, reward)
+
+                # Write new observation back into the episode.
+                single_agent_episode.set_observation(new_value=new_obs, at_index=-1)
+
+            return batch
+
+
+.. tip::
+    There is already an off-the-shelf ConnectorV2 piece available for users, which performs the task of
+    adding the `N` most recent rewards and/or `M` most recent actions to the observations:
+
+    .. code-block:: python
+
+        from ray.rllib.connectors.env_to_module.prev_actions_prev_rewards import PrevActionsPrevRewards
+
+        config.env_runners(
+            env_to_module_connector=lambda env: PrevActionsPrevRewards(n_prev_rewards=N, n_prev_actions=M),
+        )
+
+
+If you plug in this custom :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` class into your algorithm config
+(`config.env_runners(env_to_module_connector=lambda env: AddLastRewardToObservations())`),
+your model should receive observations that have the most recent reward "attached" to it.
+
+Notice that in the example here, the transformed observations were written right back into the given episodes
+and not placed into the `batch`. This strategy of writing back those data that was pulled from episodes right back
+into the same episodes makes sure that from this point on, only the changed data is visible to all subsequent components (for example
+other ConnectorV2 pieces in the same pipeline or other ConnectorV2 pipelines). It does not touch the `batch`.
+However, knowing that one of the subsequent :ref:`default env-to-module pieces <default-env-to-module-pipeline>`
+is going to do exactly that, we can defer this task (of populating the batch with our changed data) to these default pieces.
+
+The next example, however, demonstrates a specific case - observation stacking - where this strategy fails and in
+which users should instead manipulate the `batch` directly.
+
+
+Stacking the N most recent observations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If users would like to write a custom env-to-module connector that stacks the `N` most recent observations and feeds
+this stack of observations into their RLModule (for example in an attention/transformer architecture), they can
```

Comment:
Instead of "they can achieve this, too" just say "You must subclass and override..."